src/main/scala/is/hail/utils/IntervalTree.scala

```diff
@@ -37,6 +52,43 @@ case class Interval(start: Any, end: Any, includeStart: Boolean, includeEnd: Boo
 
 
   override def toString: String = (if (includeStart) "[" else "(") + start + "-" + end + (if (includeEnd) "]" else ")")
+
+  def toRegionValueInterval(pType: Type, rvb: Option[RegionValueBuilder]): RegionValueInterval = {
+    val iType = TInterval(pType)
+    val rvb2 = rvb.getOrElse(new RegionValueBuilder(Region()))
+    rvb2.start(iType)
+    rvb2.startStruct()
+    rvb2.addAnnotation(pType, start)
+    rvb2.addAnnotation(pType, end)
+    rvb2.addBoolean(includeStart)
+    rvb2.addBoolean(includeEnd)
+    rvb2.endStruct()
+    RegionValueInterval(iType, rvb2.region, rvb2.end())
+  }
+}
+
+case class RegionValueInterval(iType: TInterval, region: Region, offset: Long) extends BaseInterval {
```

Comment:
So I can't say I'm super compelled by this.  It was probably my fault for being unclear, but I think we've added similar unnecessary complexity elsewhere with IVariant, etc.  I'd just have Interval of the annotation variety, and a function `Interval.fromRegionValue(pType, region, offset)` that pulls out an Interval from a region value (which gets used in UnsafeRow.read, etc.)